abstractions/go/authentication/base_bearer_token_authentication_provider.go

```diff
@@ -40,3 +44,8 @@ func (provider *BaseBearerTokenAuthenticationProvider) AuthenticateRequest(reque
 
 	return nil
 }
+
+// GetAuthorizationTokenProvider returns the access token provider the BaseBearerTokenAuthenticationProvider class uses to authenticate the request.
+func (provider *BaseBearerTokenAuthenticationProvider) GetAuthorizationTokenProvider() AccessTokenProvider {
```

Comment:
what would need access to this?

Reply:
so people can reuse what we already built to execute arbitrary requests outside of the fluent API using the native client for example.
I don't anticipate any immediate use in any language besides JS/TS to maintain the compatibility with the existing experience without having to duplicate the whole auth stack. Added it to other languages to maintain consistency.